esy/System.ml

```diff
@@ -1,8 +1,12 @@
-type system =
+type t =
   | Darwin
   | Linux
   | Cygwin
-  | Other
+  | Other of string
```

Comment:
What other specific systems we need to know about?

Reply:
Per our discussion on Discord - we should distinguish between `Windows` and `Cygwin`. 
- `Windows` - native Windows environment (ie, someone runs `esy build` from a Windows shell)
- `Cygwin` - Cygwin environment (ie, someone running `esy` completely from Cygwin)

The `Windows` - Windows-shell case is the most important to address right now, as this is the scenario that will enable developers on Windows to produce native executables with the least friction.

Reply:
Added `Windows`.